### DISCUSSION TITLE
The "hardtime" plugin lacks the "settings" thingy

### DISCUSSION THREAD

**Discussion by refaelsh** [nix-community/nixvim#2192] [Q&A] [answered]
Here is a part of my NixOS config:
```nix
hardtime = {
  enable = true;
  disableMouse = false;
  settings = {
    showmode = false;
  };
};
```
I get the following error when rebuilding NixOS:
```
error: The option `programs.nixvim.plugins.hardtime.settings' does not exist
```

Please help :-)

**Comment by MattSturgeon** [comment]
You can find docs [here](https://nix-community.github.io/nixvim/plugins/hardtime/index.html) (and [here](https://nix-community.github.io/nixvim/plugins/hardtime/hints.html)).

Note that not all our plugins are up to date with our current best practices, i.e. freeform `settings` options. Hardtime _does_ have various options that are equivalent to `settings.*` options, and also an [extraOptions](https://nix-community.github.io/nixvim/plugins/hardtime/index.html#pluginshardtimeextraoptions) option that enables freeform definitions.

Eventually most/all of our plugins will be migrated to the new `settings` pattern though. Thanks for highlighting this one that needs some love!

**Reply by MattSturgeon** [reply, marked as answer]
In your case you can just replace `settings` with `extraOptions` for now.

**Comment by refaelsh** [comment]
> Eventually most/all of our plugins will be migrated to the new settings pattern though.

Would it be ok please, if I try and open a PR to do the migration?

**Reply by MattSturgeon** [reply]
Of course! The [CONTRIBUTING guide](https://github.com/nix-community/nixvim/blob/main/CONTRIBUTING.md) should help (if it doesn't, we'd greatly welcome any feedback on it). :rocket: 

EDIT: ah, didn't look at your name, of course you've contributed before :slightly_smiling_face:

**Comment by refaelsh** [comment]
For googlers from the future, here is the PR: https://github.com/nix-community/nixvim/pull/2193.
